### PR TITLE
Force write journal to initialize when read journal is started

### LINKED ISSUE #344: Auto initialization of tables isn't supported from the read journal side.
**Version Information**
Version of Akka.NET? 
1.5.14
Which Akka.NET Modules?
Akka.Persistence.Sql

**Describe the bug**
I am using the new LINQ to db persistence with hocon config. I am initializing my tables with  auto-initialize = true. 
But this only happens if any actors attempt to persist something. In my case I also start a read journal. Then read journal complains the tables don't exist yet. Since no events are persisted yet.  Surprisingly though the tables are created after read journal throws.

I have configured auto-initialize for the read journal as below
```hocon
akka {
    persistence {
      journal {
        plugin = "akka.persistence.journal.sql"
        sql {
           class = "Akka.Persistence.Sql.Journal.SqlWriteJournal, Akka.Persistence.Sql"
            connection-string = ${config.connection-string}
            provider-name =  "SQLite.MS"
            auto-initialize = true
        }
      }
      query.journal.sql {
        class = "Akka.Persistence.Sql.Query.SqlReadJournalProvider, Akka.Persistence.Sql"
        connection-string = ${config.connection-string}
        provider-name = "SQLite.MS"
         auto-initialize = true
      }
      snapshot-store {
        plugin = "akka.persistence.snapshot-store.sql"
        sql {
          class = "Akka.Persistence.Sql.Snapshot.SqlSnapshotStore, Akka.Persistence.Sql"
            connection-string = ${config.connection-string}
            provider-name = "SQLite.MS"
            auto-initialize = true
        }
      }
...
    
```

But I saw that auto-initialize isn't a supported config option for query.journal.sql. From the code I saw that, initilializing is done by
SqlWriteJournal. So either provide a programmatic way to force generating tables or better make sure read journal has also ability to generate the tables in case it is started before.

**To Reproduce**
Just try to use a read journal without persisting any events.

_Links to working reproductions on Github / Gitlab are very much appreciated_

**Expected behavior**
I would like readjournal not to fail with table don't exist error without persisting a dummy event.

**Actual behavior**
Read journal throws an exception when reading events. But somehow tables are created then.
```
[18:05:38 ERR] An exception occured inside SelectAsync while executing Task. Supervision strategy: Stop
System.AggregateException: One or more errors occurred. (SQLite Error 1: 'no such table: journal'.)
 ---> Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 1: 'no such table: journal'.
   at Microsoft.Data.Sqlite.SqliteException.ThrowExceptionForRC(Int32 rc, sqlite3 db)
   at Microsoft.Data.Sqlite.SqliteCommand.PrepareAndEnumerateStatements()+MoveNext()
   at Microsoft.Data.Sqlite.SqliteCommand.GetStatements()+MoveNext()
   at Microsoft.Data.Sqlite.SqliteDataReader.NextResult()
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReader(CommandBehavior behavior)
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at LinqToDB.Data.DataConnection.ExecuteReaderAsync(CommandBehavior commandBehavior, CancellationToken cancellationToken)
   at LinqToDB.Data.DataConnection.ExecuteDataReaderAsync(CommandBehavior commandBehavior, CancellationToken cancellationToken)
   at LinqToDB.Data.DataConnection.ExecuteDataReaderAsync(CommandBehavior commandBehavior, CancellationToken cancellationToken)
   at LinqToDB.Data.DataConnection.QueryRunner.ExecuteReaderAsync(CancellationToken cancellationToken)
   at LinqToDB.Linq.QueryRunner.ExecuteQueryAsync[T](Query query, IDataContext dataContext, Mapper`1 mapper, Expression expression, Object[] ps, Object[] preambles, Int32 queryNumber, Func`2 func, TakeSkipDelegate skipAction, TakeSkipDelegate takeAction, CancellationToken cancellationToken)
   at LinqToDB.Linq.QueryRunner.ExecuteQueryAsync[T](Query query, IDataContext dataContext, Mapper`1 mapper, Expression expression, Object[] ps, Object[] preambles, Int32 queryNumber, Func`2 func, TakeSkipDelegate skipAction, TakeSkipDelegate takeAction, CancellationToken cancellationToken)
   at LinqToDB.Linq.ExpressionQuery`1.GetForEachAsync(Action`1 action, CancellationToken cancellationToken)
   at LinqToDB.Linq.ExpressionQuery`1.GetForEachAsync(Action`1 action, CancellationToken cancellationToken)
   at LinqToDB.AsyncExtensions.ToListAsync[TSource](IQueryable`1 source, CancellationToken token)
   at Akka.Persistence.Sql.Query.Dao.BaseByteReadArrayJournalDao.<>c__DisplayClass6_0.<<JournalSequence>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Akka.Persistence.Sql.Extensions.ConnectionFactoryExtensions.ExecuteWithTransactionAsync[T](AkkaPersistenceDataConnectionFactory factory, IsolationLevel level, CancellationToken token, Func`3 handler)
   at Akka.Persistence.Sql.Extensions.ConnectionFactoryExtensions.ExecuteWithTransactionAsync[T](AkkaPersistenceDataConnectionFactory factory, IsolationLevel level, CancellationToken token, Func`3 handler)
   at Akka.Persistence.Sql.Extensions.ConnectionFactoryExtensions.ExecuteWithTransactionAsync[T](AkkaPersistenceDataConnectionFactory factory, IsolationLevel level, CancellationToken token, Func`3 handler)
   at Akka.Persistence.Sql.Extensions.ConnectionFactoryExtensions.ExecuteWithTransactionAsync[T](AkkaPersistenceDataConnectionFactory factory, IsolationLevel level, CancellationToken token, Func`3 handler)
   at Akka.Persistence.Sql.Query.Dao.BaseByteReadArrayJournalDao.<JournalSequence>b__6_0(<>f__AnonymousType4`3 input)
   --- End of inner exception stack trace ---
   at Akka.Actor.PipeToSupport.PipeTo[T](T
````

**Environment**
Linux, docker .NET 8
Are you running on Linux? Windows? Docker? Which version of .NET?

### PR DESCRIPTION
**Pull request by Aaronontheweb**
## Changes

close #344

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [x] Design discussion issue #344
* [x] Changes in public API reviewed, if any.

### CODE REVIEW

File: src/Akka.Persistence.Sql.Tests.Common/Query/Bugfix344Spec.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+using Akka.Streams;
+using Akka.Streams.TestKit;
+using FluentAssertions.Extensions;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Akka.Persistence.Sql.Tests.Common.Query
+{
+    public abstract class Bugfix344Spec<T> : Akka.TestKit.Xunit2.TestKit, IAsyncLifetime where T : ITestContainer
```
Comment (Aaronontheweb) [inline src/Akka.Persistence.Sql.Tests.Common/Query/Bugfix344Spec.cs:23]: Abstract spec to validate that this issue has been resolved, and that the `ReadJournal` will automatically start up the write journal and wait for it to initialize.

File: src/Akka.Persistence.Sql/Query/SqlReadJournal.cs [hunk 2]
```diff
@@ -59,6 +59,11 @@ public class SqlReadJournal :
             _system = system;
 
             var connFact = new AkkaPersistenceDataConnectionFactory(_readJournalConfig);
+            
+            // Fix for https://github.com/akkadotnet/Akka.Persistence.Sql/issues/344
+            var writeJournal = Persistence.Instance.Apply(system).JournalFor(writePluginId);
```
Comment (Aaronontheweb) [inline src/Akka.Persistence.Sql/Query/SqlReadJournal.cs:59]: Start the `SqlWriteJournal` if it has not been started already - technically the `_eventAdapters = Persistence.Instance.Apply(system).AdaptersFor(writePluginId);` also does this implicitly.

File: src/Akka.Persistence.Sql/Query/SqlReadJournal.cs [hunk 3]
```diff
@@ -59,6 +59,11 @@ public class SqlReadJournal :
...
+            // we want to block, we want to crash if the journal is not available
+            var started = writeJournal.Ask<Initialized>(IsInitialized.Instance, TimeSpan.FromSeconds(5)).Result;
```
Comment (Aaronontheweb) [inline src/Akka.Persistence.Sql/Query/SqlReadJournal.cs:61]: Block and wait until startup is complete - this should cover any cases of this issue where queries still get executed before the write journal fully finishes initializing.

**Review by Aaronontheweb — COMMENTED** [review]
Detailed my changes

**Review by Arkatufus — APPROVED** [review]
LGTM

**Comment by Aaronontheweb** [comment]
Got the error to occur:

```
Failed: Expected a message of type Akka.Streams.TestKit.TestSubscriber+OnComplete, but received {TestSubscriber.OnError(One or more errors occurred. (42P01: relation "journal" does not exist


POSITION: 31))} (type Akka.Streams.TestKit.TestSubscriber+OnError) instead  from [akka://test/user/StreamSupervisor-1252/Flow-0-0-unknown-operation#440889324]
```

**Pull request merged**